### PR TITLE
#5281 Add private key error handling on my_key_update.ts

### LINKED ISSUE #5281: Error: Key type is unknown, expecting OpenPGP or x509 S/MIME in key_update frame
I got 33 of these errors reported, sounds like something that should be handled to show the user a modal with some help, but not reported as an exception.

33 | title: Error                                         message: Key type is unknown, expecting OpenPGP or x509 S/MIME                                         url: chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/chrome/settings/modules/my_key_update.htm?acctEm                                         line: 126                                         column: 15                                         environment: chrome:ex:prod                                         build type: consumer | Error: Key type is unknown, expecting OpenPGP or x509 S/MIME                                at KeyUtil.parseMany (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/js/common/core/crypto/key.js:126:15)                                at KeyUtil.parse (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/js/common/core/crypto/key.js:105:36)                                at MyKeyUpdateView.updatePrivateKeyHandler  (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/chrome/settings/modules/my_key_update.js:87:50)                                at MyKeyUpdateView.<anonymous>  (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/chrome/settings/modules/my_key_update.js:55:101)                                at cbWithErrsHandled (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/js/common/browser/ui.js:102:48)                                at HTMLSpanElement.<anonymous>  (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/js/common/browser/ui.js:140:21)                                at HTMLSpanElement.dispatch (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/lib/jquery.min.js:2:39997)                                at v.handle (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/lib/jquery.min.js:2:37968)                                at <async> parseMany (/extension/js/common/core/crypto/key.ts:191:28)                                at <async> parse (/extension/js/common/core/crypto/key.ts:182:24)                                at <async> updatePrivateKeyHandler (/extension/chrome/settings/modules/my_key_update.ts:103:38)                                                        ### Catch.reportErr calling stack ###                            #     at Catch.formatExceptionForReport  (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/js/common/platform/catch.js:273:90)                            #     at Catch.onErrorInternalHandler  (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/js/common/platform/catch.js:81:29)                            #     at Catch.reportErr (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/js/common/platform/catch.js:97:18)                            #     at Object._dispatchErr (chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/js/common/browser/ui.js:90:19)                            #     at chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/js/common/browser/ui.js:105:43                            #                             # url:  chrome-extension://bnjglocicdkmhmoohhfkfkbbkejdhdgc/chrome/settings/modules/my_key_update.htm?acctEmail=[SCRUBBED]&fingerprint=
-- | -- | --

### PR DESCRIPTION
**Pull request by martgil**
This PR adds a private key error handling on my_key_update.ts.

close #5281 

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

**Comment by martgil** [comment]
@sosnovsky ready for review. thank you.

File: extension/chrome/elements/shared/key_errors.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import { Settings } from '../../../js/common/settings.js';
+import { UserAlert, KeyCanBeFixed } from '../../../js/common/ui/key-import-ui.js';
+import { SetupOptions, SetupView } from '../../settings/setup.js';
+
+export class KeyErrors {
+  protected fesUrl: string;
+  private readonly acctEmail: string;
+  private readonly parentTabId: string;
+  private readonly clientConfiguration: ClientConfiguration;
+  private readonly setupView?: SetupView;
```
Comment (sosnovsky) [inline extension/chrome/elements/shared/key_errors.ts:21]: It seems these properties are used only in this class, so we don't need `protected` and `readonly`, they all can be just `private`.

Also let's simplify properties and constructor code by using [parameter properties](https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties):
```ts
public constructor(
    private fesUrl: string,
    private acctEmail: string,
    private parentTabId: string,
    private clientConfiguration: ClientConfiguration,
    private setupView?: SetupView
) {}
```

**Review by sosnovsky — COMMENTED** [review]
Looks good, let's also add test which checks if invalid key error is properly handled on key update

File: extension/chrome/elements/shared/key_errors.ts [hunk 2]
```diff
@@ -23,6 +23,7 @@ export class KeyErrors {
   ) {}
 
   public handlePrivateKeyError = async (exception: unknown, origPrv: Key, setupOptions?: SetupOptions) => {
+    console.log('test');
```
Comment (martgil) [inline extension/chrome/elements/shared/key_errors.ts]: just added and remove this temporarily to somehow fixed CI test not working correctly aftering clicking 'rerun' which doesn't produce the desired effect.

File: extension/chrome/elements/shared/key_errors.ts (hunk 1 above)
Comment (martgil) [reply extension/chrome/elements/shared/key_errors.ts]: Thank you very much. I have learned more here. I have removed the unused properties and converted them into parameter properties.

**Comment by martgil** [comment]
ready for review again. thank you 😅.

**Review by sosnovsky — COMMENTED** [review]
All good, but you've probably missed my comment on previous review - please also add test for checking if errors are handled correctly on private key update, thanks!

**Review by tomholub — APPROVED** [review]
I'll merge it since I'm doing a release soon, can do test in a follow-up PR :+1:

**Comment by ioanmo226** [comment]
@tomholub Do you plan to release without fixing this? Some users (especially for those with slow internet connection)might think extension is broken.
https://github.com/FlowCrypt/flowcrypt-browser/issues/5333

**Pull request merged**